WordPress/Classes/ViewRelated/Blog/Blog List/BlogListViewController.m

```diff
@@ -792,15 +818,23 @@ - (void)setSelectedBlog:(Blog *)selectedBlog
 
 - (void)setSelectedBlog:(Blog *)selectedBlog animated:(BOOL)animated
 {
+    if ([Feature enabled:FeatureFlagNewNavBarAppearance]) {
+        if (self.blogSelected != nil) {
+            self.blogSelected(self, selectedBlog);
+        } else {
+            // The site picker without a site-selection callback makes no sense.  We'll dismiss the VC to keep
+            // the app running, but the user won't be able to switch sites.
+            DDLogError(@"There's no site-selection callback assigned to the site picker.");
+            [self dismissViewControllerAnimated:animated completion:nil];
+        }
+
+        return;
+    }
+    
     if (selectedBlog != _selectedBlog || !_blogDetailsViewController) {
         _selectedBlog = selectedBlog;
         self.blogDetailsViewController = [self makeBlogDetailsViewController];
         self.blogDetailsViewController.blog = selectedBlog;
-
-        if (![self splitViewControllerIsHorizontallyCompact]) {
```

Comment:
I think we may need to re-add this, as I'm occasionally seeing an empty detail view on iPad:

![IMG_0572](https://user-images.githubusercontent.com/4780/110018731-855f3700-7d1f-11eb-8162-3b5c6c4a5fc8.PNG)